bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinBaseHandler.java

```diff
@@ -131,6 +132,8 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             }
             logger.debug("Received command ({}) of wrong type for thing '{}' on channel {}", command,
                     thing.getUID().getAsString(), channelUID.getId());
+        } catch (IllegalArgumentException ex) {
+            logger.warn("Illegal argument for channel {}: {}. {}", channelUID.getId(), command.toString(), ex.getMessage());
```

Comment:
Instead of catching here, this should be caught around the Enum.valueOf call that is throwing this.

Reply:
Catching it around Enum.valueOf would duplicate the same code across 5 different instances at the moment, and possible future additions. In my opinion that is not an elegant/clean approach. Yes, catching it here could potentially catch IllegalArgumentException thrown by things other than enum.ValueOf although I can't think of what at the moment, and that's why I included ex.getMessage() in the logging for clarifications.

Is there any specific reasons that it should be caught around the enum.valueOf, and causing code duplication?

Reply:
While I do agree that catching IllegalArgumentException is a elegant/clean approach to solve this problem I still think that it isn't the correct approach. Catching IllegalArgumentException here simply casts too wide of a net and from a code readability/maintainability standpoint it isn't apparent at all what would throw such an exception at this point in the code or if the way it is handled is even correctly. 
Don't get me wrong, I do think you've implemented the correct error handling at the moment but there may be a point in the future where that is not the case.

For readability/maintainability you should catch the IllegalArgumentException as soon as possible so that it is easy to identify from the code what would throw it and why it would be thrown. Now if it important that a malformed enum exception gets propagated all the way to the top like this, feel free to create your own type of exception to represent this and throw that instead. IllegalArgumentException is too broad to be allowed to be propagated beyond the caller of the method that throws it without the exception itself becoming essentially unhandleable.

If you want to reduce possible duplicate code you can make your own version of the `valueOf` that takes an Enum as an argument and use that instead.

Reply:
Point taken. I have moved the exception handling.